src/main/java/de/srendi/advancedperipherals/common/util/inventory/BasicCraftJob.java

```diff
@@ -0,0 +1,206 @@
+package de.srendi.advancedperipherals.common.util.inventory;
+
+import dan200.computercraft.api.lua.LuaFunction;
+import dan200.computercraft.api.peripheral.IComputerAccess;
+import de.srendi.advancedperipherals.common.util.StatusConstants;
+import net.minecraft.world.level.Level;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public abstract class BasicCraftJob {
+
+    protected static final String EVENT = "_crafting";
+    private static final int JOB_DONE_PURGE_TIME = 5 * 60 * 1000;
+
+    public static final AtomicLong ID_SEQ = new AtomicLong();
+
+    protected final long id = ID_SEQ.incrementAndGet();
+    protected final IComputerAccess computer;
+    protected final String eventName;
+    protected final long amount;
+    protected final Level world;
+
+    protected boolean startedCrafting = false;
+    protected boolean startedCalculation = false;
+    protected boolean calculationNotSuccessful = false;
+    protected boolean errorOccurred = false;
+    protected boolean isJobDone = false;
+    protected long jobDoneTime = 0;
+    protected boolean isJobCanceled = false;
+    protected String debugMessage = null;
+
+    public BasicCraftJob(IComputerAccess computer, String eventName, Level world, long amount) {
+        this.computer = computer;
+        this.eventName = eventName;
+        this.world = world;
+        this.amount = amount;
+    }
+
+    @LuaFunction
+    public final long getId() {
+        return id;
+    }
+
+    protected abstract boolean isJobDone();
+
+    @LuaFunction
+    public final boolean isDone() {
+        return isJobDone();
+    }
+
+    protected abstract boolean isJobCanceled();
+
+    @LuaFunction
+    public final boolean isCanceled() {
+        return isJobCanceled();
+    }
+
+    @LuaFunction
+    public final boolean isCraftingStarted() {
+        return startedCrafting;
+    }
+
+    @LuaFunction
+    public final boolean isCalculationStarted() {
+        return startedCalculation;
+    }
+
+    @LuaFunction
+    public final boolean isCalculationNotSuccessful() {
+        return calculationNotSuccessful;
+    }
+
+    @LuaFunction
+    public final boolean hasErrorOccurred() {
+        return errorOccurred;
+    }
+
+    @LuaFunction
+    public final String getDebugMessage() {
+        return debugMessage;
+    }
+
+    @LuaFunction(value = "getRequestedItem")
```

Comment:
I believe you can ignore the `value =` if the only value is presenting

Reply:
I actually didn't noticed that haha